src/app/services/baw-api/baw-api.service.ts

```diff
@@ -145,13 +150,14 @@ export class BawApiService<
     @Inject(IS_SERVER_PLATFORM) protected isServer: boolean,
     protected manager: HttpCacheManager,
     protected http: HttpClient,
-    protected injector: Injector,
     protected session: BawSessionService,
     protected notifications: ToastrService,
-    @Inject(CACHE_SETTINGS) private cacheSettings: CacheSettings
+    @Inject(CACHE_SETTINGS) private cacheSettings: CacheSettings,
+    @Inject(ASSOCIATION_INJECTOR.token) protected associationInjector: any,
+    @Optional() @Inject(BAW_SERVICE_OPTIONS) private options: BawServiceOptions
```

Comment:
ok, flatten options with default in the ctor and store the merged result

- default options: static and frozen
- instance options: Object.assign({},  defaultOptions, options)
- and ideally options that are set instance wide are not configurable in the methods

Now that we have configurable services, do we need the options configurable from individual methods? Where is disableNotificaitons used elsewhere?

Reply:
`disableNotification` is used in the `security.service` when we try to get the currently logged in user through a GET request.
Obviously in this case, we don't want to raise an error to the user if the user is not logged in, causing fetching the current user to fail.

I don't expect merging injected and instance options to break anything

Reply:
if that's the only other case, could we provide a service configured in the correct way and remove the method overload?

Reply:
Associated issue: https://github.com/QutEcoacoustics/workbench-client/issues/2166